src/swsssdk/interface.py

```diff
@@ -159,206 +159,210 @@ def __init__(self, **kwargs):
         # notifications for each client
         self.keyspace_notification_channels = DBRegistry()
 
-    def connect(self, db_id, retry_on=True):
+    def connect(self, db_id, db_name, retry_on=True):
         """
         :param db_id: database id to connect to
+        :param db_name: database name to connect to
         :param retry_on: if ``True`` -- will attempt to connect continuously.
         if ``False``, only one attempt will be made.
         """
         if retry_on:
-            self._persistent_connect(db_id)
+            self._persistent_connect(db_id, db_name)
         else:
-            self._onetime_connect(db_id)
+            self._onetime_connect(db_id, db_name)
 
-    def _onetime_connect(self, db_id):
+    def _onetime_connect(self, db_id, db_name):
         """
         Connect to database id.
         """
         if db_id is None:
             raise ValueError("No database ID configured for '{}'".format(db_id))
 
+        if db_name is None:
+            raise ValueError("No database Name configured for '{}'".format(db_name))
+
         client = redis.StrictRedis(db=db_id, **self.redis_kwargs)
```

Comment:
>redis.StrictRedis [](start = 17, length = 17)

If `self.redis_clients` already has this StrictRedis, we don't need to create a new one.

Reply:
it is a day 1 behavior, we might consider update all other previous release as well. Anyway we can fix it in dev-master for now. code updated.